services/hanke-service/src/main/kotlin/fi/hel/haitaton/hanke/hakemus/HakemusService.kt

```diff
@@ -911,10 +918,67 @@ class HakemusService(
         if (updatedApplicationEntity.hanke.generated) {
             updatedApplicationEntity.hanke.nimi = request.name
         }
+        updateTormaystarkastelut(updatedApplicationEntity)
         sendApplicationNotifications(updatedApplicationEntity, originalContactUserIds, userId)
         return hakemusRepository.save(updatedApplicationEntity)
     }
 
+    /** Calculate the traffic nuisance indexes for each work area of an application. */
+    private fun updateTormaystarkastelut(hakemusEntity: HakemusEntity) {
+        if (hakemusEntity.hakemusEntityData is KaivuilmoitusEntityData) {
+            val hakemusEntityData = hakemusEntity.hakemusEntityData as KaivuilmoitusEntityData
+            if (hakemusEntityData.startTime != null && hakemusEntityData.endTime != null) {
+                hakemusEntity.hakemusEntityData =
+                    hakemusEntityData.copy(
+                        areas =
+                            hakemusEntityData.areas?.map { area ->
+                                updateTormaystarkastelutForArea(
+                                    area,
+                                    hakemusEntityData.startTime.toLocalDate(),
+                                    hakemusEntityData.endTime.toLocalDate(),
+                                )
+                            },
+                    )
+            }
+        }
+    }
+
+    private fun updateTormaystarkastelutForArea(
+        area: KaivuilmoitusAlue,
+        startDate: LocalDate,
+        endDate: LocalDate
+    ) =
+        area.copy(
+            tyoalueet =
+                area.tyoalueet.map { tyoalue ->
+                    updateTormaystarkasteluForTyoalue(
+                        tyoalue,
+                        startDate,
+                        endDate,
+                        area.kaistahaitta,
+                        area.kaistahaittojenPituus,
+                    )
+                },
+        )
+
+    private fun updateTormaystarkasteluForTyoalue(
+        tyoalue: Tyoalue,
+        startDate: LocalDate,
+        endDate: LocalDate,
+        kaistahaitta: VaikutusAutoliikenteenKaistamaariin,
+        kaistapituushaitta: AutoliikenteenKaistavaikutustenPituus,
+    ) =
+        tyoalue.copy(
+            tormaystarkasteluTulos =
+                tormaystarkasteluLaskentaService.calculateTormaystarkastelu(
+                    FeatureCollection().apply {
+                        add(Feature().apply { this.geometry = tyoalue.geometry })
+                    },
+                    daysBetween(startDate, endDate)!!,
```

Comment:
Overloading `daysBetween()` with a non-null version would eliminate the need for `!!`:
```kotlin
fun daysBetween(startDate: Temporal, endDate: Temporal): Int =
    ChronoUnit.DAYS.between(startDate, endDate).toInt() + 1
```

Reply:
Added the overloading.